tests/test_metric_store.py

```diff
@@ -1,72 +1,101 @@
 """XAIN FL tests for metric store"""
-# pylint: disable=redefined-outer-name
+import json
 from unittest import mock
 
 from influxdb import InfluxDBClient
-import numpy as np
 import pytest
 
 from xain_fl.config import MetricsConfig
-from xain_fl.coordinator.metrics_store import (
-    MetricsStore,
-    MetricsStoreError,
-    transform_metrics_to_influx_data_points,
-)
+from xain_fl.coordinator.metrics_store import MetricsStore, MetricsStoreError
 
 
 @pytest.fixture()
-def metrics_sample():
+def metrics_sample_empty():
     """Return a valid metric object."""
-    return {
-        "metric_1": np.array([0.2, 0.44]),
-        "metric_2": np.array([0.99, 0.55]),
-    }
+    return json.dumps([])
 
 
-def test_transform_data(metrics_sample):
-    """Check that a metric object is correctly transformed into the influx data point structure."""
-    actual_data_points = transform_metrics_to_influx_data_points(
-        "participant_id", metrics_sample
+@pytest.fixture()
+def metrics_sample_invalid():
+    """Return a valid metric object."""
+    return json.dumps(
+        [
+            {
+                "measurement": "CPU utilization",
+                "time": "00:00:00",
+                "tags": {"id": "127.0.0.1:1345"},
+            }
+        ]
     )
-    expected_data_points = [
-        {
-            "measurement": "participant.ai.metric_1",
-            "tags": {"id": "participant_id"},
-            "fields": {"metric_1": "0.20000000"},
-        },
-        {
-            "measurement": "participant.ai.metric_1",
-            "tags": {"id": "participant_id"},
-            "fields": {"metric_1": "0.44000000"},
-        },
-        {
-            "measurement": "participant.ai.metric_2",
-            "tags": {"id": "participant_id"},
-            "fields": {"metric_2": "0.99000000"},
-        },
-        {
-            "measurement": "participant.ai.metric_2",
-            "tags": {"id": "participant_id"},
-            "fields": {"metric_2": "0.55000000"},
-        },
-    ]
-
-    for (actual_dp, expected_dp) in zip(actual_data_points, expected_data_points):
-        assert actual_dp["measurement"] == expected_dp["measurement"]
-        assert actual_dp["tags"] == expected_dp["tags"]
-        assert actual_dp["fields"] == expected_dp["fields"]
-
-    assert actual_data_points[0]["time"] == actual_data_points[2]["time"]
-    assert actual_data_points[1]["time"] == actual_data_points[3]["time"]
+
+
+@mock.patch.object(InfluxDBClient, "write_points", return_value=True)
+def test_valid_metric(
+    write_points_mock, metrics_sample,
+):  # pylint: disable=redefined-outer-name,unused-argument
+    """Check that raised exceptions of the write_points method are caught in the write_metrics
```

Comment:
this description doesn't match the content of the test

Reply:
Fixed